src/Raven.Server/Documents/SubscriptionStorage.cs

```diff
@@ -396,29 +398,60 @@ private unsafe DynamicJsonValue ExtractSubscriptionConfigValue(TableValueReader
             }
         }
 
-        // ReSharper disable once ClassNeverInstantiated.Local
-        public class Schema
+        public void GetRunningSubscriptionConnectionHistory(BlittableJsonTextWriter writer, DocumentsOperationContext context, long subscriptionId)
         {
-            public static readonly string IdsTree = "SubscriptionsIDs";
-            public static readonly string SubsTree = "Subscriptions";
-            public static readonly Slice Id = Slice.From( StorageEnvironment.LabelsContext, "Id", ByteStringType.Immutable);
+            SubscriptionState subscriptionState;
+            if (!_subscriptionStates.TryGetValue(subscriptionId, out subscriptionState)) return;
+
+            var subscriptionConnection = subscriptionState.Connection;
+            if (subscriptionConnection == null) return;
 
-            public static class SubscriptionTable
+            using (var tx = _environment.ReadTransaction())
             {
-#pragma warning disable 169
-                public static readonly int IdIndex = 0;
-                public static readonly int CriteriaIndex = 1;
-                public static readonly int AckEtagIndex = 2;
-                public static readonly int TimeOfSendingLastBatch = 3;
-                public static readonly int TimeOfLastActivityIndex = 4;
-#pragma warning restore 169
+                var subscriptionData =
+                    ExtractSubscriptionConfigValue(GetSubscriptionConfig(subscriptionId, tx), context);
+                WriteSubscriptionStateData(subscriptionState, subscriptionData, true);
+
+                context.Write(writer, subscriptionData);
+                writer.Flush();
             }
         }
+        public long GetRunningCount()
+        {
+            return _subscriptionStates.Count(x=>x.Value.Connection!=null);
+        }
 
 
         public StorageEnvironment Environment()
         {
             return _environment;
         }
+        public long GetAllSubscriptionsCount()
+        {
+            using (var tx = _environment.WriteTransaction())
+            {
+                var table = new Table(_subscriptionsSchema, Schema.SubsTree, tx);
+                return table.NumberOfEntries;
+            }
+        }
+    }
+
+    // ReSharper disable once ClassNeverInstantiated.Local
+    public class Schema
```

Comment:
make this a static class, and what is the meaning of `Schema` in this context? It is too generic a word